MathLib/LinAlg/UnifiedMatrixSetters.cpp

```diff
@@ -83,25 +83,35 @@ double norm(PETScVector const& x)
 void setVector(PETScVector& v,
                std::initializer_list<double> values)
 {
-    (void) v; (void) values;
-    // TODO implement
-}
+    std::vector<double> const vals(values);
+    std::vector<PETScVector::IndexType> idcs(vals.size());
+    std::iota(idcs.begin(), idcs.end(), 0);
 
+    v.set(idcs, vals);
+}
 
 void setMatrix(PETScMatrix& m,
                PETScMatrix::IndexType const rows,
                PETScMatrix::IndexType const cols,
                std::initializer_list<double> values)
 {
-    (void) m; (void) rows; (void) cols; (void) values;
-    // TODO implement
+    m.setZero();
+    addToMatrix(m, rows, cols, values);
```

Comment:
First of all, those methods are currently intended for use in tests. And I think, there is the only place where they are used.
Yes, matrix dims can be set independently.
The potential for misuse is currently restricted to tests.
Since currently only tests are affected, the check is not time-critical. However, it is common that bounds checks are disabled in release mode. So I think assertions suffice.
I'll add checks for the matrix bounds, then the size arguments can also be dropped.